### PR TITLE
Mds 3885 update emli contact info

### PR DESCRIPTION
**Pull request by mmumby**
# Main
- Add in Core the ability to perform CRUD operations for EMLI users 
- MineSpace change from reading users (/src/constants/contacts.js) to fetch users from database.
- Include regional , major mines and general contacts.

# Other
- Creation of new table called EMLI_contact to substitute regional_contact table
- Seed data in new table

# How to test
In Core, a user can be added/edited under the Admin tab: "MineSpace EMLI Contacts"
Screen Shot 2021-12-01 at 11 47 43 AM

- Ensure generated documents are pulling the correct information from regional office:
Screen Shot 2021-12-01 at 11 53 05 AM
Screen Shot 2021-12-01 at 11 53 15 AM

In MineSpace, the contacts should be read as follow:
- Regional mine: Regional contacts based on the region mine are added and general contacts
image

- Major mine: Major mine contacts based on the region
image

# Notes
Screen Shot 2021-12-01 at 11 49 26 AM
Screen Shot 2021-12-01 at 11 49 44 AM
Screen Shot 2021-12-01 at 11 51 26 AM

Screen Shot 2021-12-01 at 11 49 03 AM

What contacts appear on each region: 
https://miro.com/app/board/uXjVOebUdk4=/
https://bcmines.atlassian.net/jira/software/c/projects/MDS/boards/1?modal=detail&selectedIssue=MDS-3885

### CODE REVIEW

File: services/core-api/app/api/EMLI_contacts/resources/EMLI_contact.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from app.api.utils.access_decorators import requires_role_edit_emli_contacts, requires_role_mine_admin, requires_any_of, VIEW_ALL, MINESPACE_PROPONENT
+from app.api.EMLI_contacts.response_models import EMLI_CONTACT_MODEL
+from app.api.EMLI_contacts.models.EMLI_contact import EMLIContact
+from app.api.EMLI_contacts.models.EMLI_contact_type import EMLIContactType
+
+
+class EMLIContactResource(Resource, UserMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('first_name', type=str, trim=True, help='EMLI First name', location='json')
+    parser.add_argument('last_name', type=str, help='EMLI Last name.', location='json')
```
Comment (BenHarrington) [inline services/core-api/app/api/EMLI_contacts/resources/EMLI_contact.py:16]: Is there a reason this field is not trimmed like the first_name?

File: services/core-api/app/api/EMLI_contacts/resources/EMLI_contact_list.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    parser.add_argument(
+        'emli_contact_type_code',
+        type=str,
+        trim=True,
+        help='EMLI contact type code',
+        location='json')
+    parser.add_argument(
+        'mine_region_code', type=str, trim=True, help='EMLI mine region code', location='json')
+    parser.add_argument('first_name', type=str, trim=True, help='EMLI First name', location='json')
+    parser.add_argument('last_name', type=str, help='EMLI Last name.', location='json')
```
Comment (BenHarrington) [inline services/core-api/app/api/EMLI_contacts/resources/EMLI_contact_list.py:24]: Same as previous comment(services/core-api/app/api/EMLI_contacts/resources/EMLI_contact.py), add trim=True for consistency or ditch it altogther.

File: services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py [hunk 3]
```diff
@@ -94,7 +95,8 @@ class StaticContentResource(Resource):
     )
     @requires_role_view_all
     def get(self):
-        content_json = cache.get(STATIC_CONTENT_KEY)
+        # content_json = cache.get(STATIC_CONTENT_KEY)
+        content_json = {}
```
Comment (BenHarrington) [inline services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py:99]: Was this accidentally left in, or was it intended?

**Review by BenHarrington — CHANGES_REQUESTED** [review]

File: services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py (hunk 3 above)
Comment (mmumby) [reply services/core-api/app/api/exports/static_content/resources/core_static_content_resource.py:99]: that was a mistake, good catch - updated.

**Comment by BenHarrington** [comment]
Testing on /admin/minespace-emli-contact-management, I just noticed that the Major or Regional Contact field shows "Reginal Mine", small typo there.

File: services/core-web/src/components/admin/contacts/EMLIContacts/EMLIContactsTable.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    dataIndex: "mine_region_code",
+    sorter: nullableStringSorter("mine_region_code"),
+    render: (text) => <div title="Region">{regionHash[text] || Strings.EMPTY_FIELD}</div>,
+  },
+  {
+    title: "Major or Regional Contact",
+    dataIndex: "is_major_mine",
+    sorter: (a, b) => (a.is_major_mine > b.is_major_mine ? -1 : 1),
+    render: (text) => (
+      <div title="Major or Regional Contact">{text ? "Major Mine" : "Reginal Mine"}</div>
```
Comment (BenHarrington) [inline services/core-web/src/components/admin/contacts/EMLIContacts/EMLIContactsTable.js:43]: As mentioned in a general comment, small typo here.

**Review by BenHarrington — CHANGES_REQUESTED** [review]

File: services/core-api/app/api/EMLI_contacts/resources/EMLI_contact.py (hunk 1 above)
Comment (mmumby) [reply services/core-api/app/api/EMLI_contacts/resources/EMLI_contact.py]: Added trim=True to last_name

File: services/core-api/app/api/EMLI_contacts/resources/EMLI_contact_list.py (hunk 2 above)
Comment (mmumby) [reply services/core-api/app/api/EMLI_contacts/resources/EMLI_contact_list.py]: Added trim=True to last_name

File: services/core-web/src/components/admin/contacts/EMLIContacts/EMLIContactsTable.js (hunk 4 above)
Comment (mmumby) [reply services/core-web/src/components/admin/contacts/EMLIContacts/EMLIContactsTable.js]: ah thank you. Fixed spelling

**Comment by BenHarrington** [comment]
Two small issues I have found since the last update:

If you try to create a contact without an email the POST fails even though the form does not indicate email is required
Completing the POST with an email included results in a "AttributeError: 'NoneType' object has no attribute 'emli_contact_type_code'" error from the API. That field is included in the request but may be accessed incorrectly in the request handler on the API.

**Comment by mmumby** [comment]
> Two small issues I have found since the last update:
> 
> 1. If you try to create a contact without an email the POST fails even though the form does not indicate email is required
> 2. Completing the POST with an email included results in a "AttributeError: 'NoneType' object has no attribute 'emli_contact_type_code'" error from the API. That field is included in the request but may be accessed incorrectly in the request handler on the API.

1. Fixed
2. I was not able to recreate this issue, can we take a look together after standup?

**Pull request closed without merging**